idr/connections.py

```diff
@@ -71,9 +79,26 @@ def connection(host=None, user=None, password=None, port=None, verbose=1):
         kwargs['port'] = port
 
     c = omero.client(**kwargs)
-    c.enableKeepAlive(300)
-    c.createSession(user, password)
-    conn = BlitzGateway(client_obj=c)
+
+    # If omero.user and omero.pass were included in omero-client.json
+    # they can optionally be omitted in createSession
+    create_session_args = []
+    if user:
+        create_session_args.append(user)
+        if password:
+            create_session_args.append(password)
+
+    try:
+        c.createSession(*create_session_args)
+        c.enableKeepAlive(300)
+        conn = BlitzGateway(client_obj=c)
+    except omero.ClientError as e:
+        if re.match(r'\w+://', host):
```

Comment:
Is this intending to capture all ice protocols and not retry? Can't host be either the raw host or a http:// configuration URL?

Reply:
The idea is to only retry if the bare host `idr.openmicroscopy.org` was given. In all other cases the ice protocol is explicit, either in the `host` or in the configuration returned by the http url which currently returns `ssl` but in future we should change this to `wss`, thus making this logic mostly redundant.

Reply:
Makes sense. That is quite clear. Might be worth stating in the docks. Otherwise, :+1:

Reply:
Done